code/game/machinery/clonepod.dm

```diff
@@ -28,367 +61,500 @@ GLOBAL_LIST_INIT(cloner_biomass_items, list(\
+			reagents.del_reagent(R.id)
+			reagents.update_total()
+			atom_say("Purged contaminant from chemical storage.")
 
-//Grow clones to maturity then kick them out. FREELOADERS
-/obj/machinery/clonepod/process()
+	//Take in biomass. Mostly copied from the old cloning code
 	var/show_message = FALSE
 	for(var/obj/item/item in range(1, src))
-		if(is_type_in_list(item, GLOB.cloner_biomass_items))
+		if(is_type_in_list(item, VALID_BIOMASSABLES) && (biomass + MEAT_BIOMASS_VALUE <= biomass_storage_capacity))
 			qdel(item)
-			biomass += BIOMASS_BASE_AMOUNT
+			biomass += MEAT_BIOMASS_VALUE
 			show_message = TRUE
 	if(show_message)
-		visible_message("[src] sucks in and processes the nearby biomass.")
+		visible_message("<span class='notice'>[src] sucks in nearby biomass.</span>")
+
+	//If we're cloning someone, we haven't generated a list of limbs to grow, and we're before any possibility of not having any limbs left to grow.
+	if(currently_cloning && !length(limbs_to_grow) && clone_progress < 20)
+		for(var/limb in desired_data.limbs)
+			if(desired_data.limbs[limb][4])
+				continue //We're not growing this limb, since in the desired state it's missing.
+
+			var/obj/item/organ/external/limb_typepath = patient_data.genetic_info.species.has_limbs[limb]["path"]
+			if(initial(limb_typepath.vital)) //I hate everything about this check, but it sees if the current organ is vital..
+				continue //and continues if it is, since the proc that creates the clone mob will make these all at once.
+			var/parent_organ_is_limb = FALSE
+			for(var/organ in desired_data.organs)
+				var/obj/item/organ/external/organ_typepath = patient_data.genetic_info.species.has_organ[organ]
+				if(!initial(organ_typepath.vital)) //I hate this check too. We loop through all the organs the cloned species should have.
+					continue //If it's not a vital organ, continue looping through the organs.
+				if(initial(organ_typepath.parent_organ) == limb)
+					parent_organ_is_limb = TRUE //If it's a vital organ, and belongs to the current limb, we don't want this limb.
+					break
+			if(parent_organ_is_limb)
+				continue
+			limbs_to_grow += limb //It's not supposed to be missing and it's not vital - so we'll be growing it.
+		shuffle(limbs_to_grow)
+
+	if(clone)
+		clone.Weaken(4 SECONDS) //make sure they stay in the pod
+		clone.setOxyLoss(0) //..and alive
+
+	//Actually grow clones (this is the fun part of the proc!)
+	if(currently_cloning)
+		clone_progress += speed_modifier
+		switch(clone_progress)
+			if(0 to 10)
+				desc_flavor = "You see muscle quickly growing on a ribcage and skull inside [src]."
+				return
+			if(11 to 90)
+				if(!clone)
+					create_clone()
+					return
+
+				if(clone.cloneloss >= 25)
+					clone.adjustCloneLoss(-2)
+					return
+
+				if(!current_limb)
+					if(!length(limbs_to_grow)) //if there's no current limb, we have no limbs left to grow, and there's <25 clone damage, we don't have anything left to do in this 'chunk' of clone_progress
```

Comment:
```suggestion
				if(!current_limb)  // If there's no current limb and
					if(!length(limbs_to_grow)) // no limbs left to grow, we don't have anything left to do in this 'chunk' of clone_progress
```
Might be best to make this comment a bit smaller + split up, so you don't have to scroll a metre away to read it